spyder/plugins/projects/api.py

```diff
@@ -94,38 +94,67 @@ def get_recent_files(self):
 
         return list(OrderedDict.fromkeys(recent_files))
 
-    def get_plugin(self, plugin_name):
-        """
-        Return a plugin by unique `plugin_name`.
-        """
-        return self.projects_plugin.get_plugin(plugin_name)
-
     # --- API
     # ------------------------------------------------------------------------
     @staticmethod
-    def get_name(self):
+    def get_name():
         """
         Provide a human readable version of NAME.
         """
         raise NotImplementedError("Must implement a `get_name` method!")
 
+    @staticmethod
+    def validate_name(path, name):

```

Comment:
Why is it necessary to validate the project's name?

Reply:
@nerohmot needs this.

Reply:
a valid name for python files does include the underscore `_`, however we don't want the **project name** to include an underscore in the project name, as the resulting test-programs (in the project) all start with the project name (it is always something relatively small like `CTCA`, `HATB`, `MN99254`) and we use underscores to further differentiate in a structured way.

When the user has the ability to use underscores in the project name, he would be encouraged to start some kind of differentiation of his own which might confuse the system itself (or attached tools) and if not, will make confusing test program names for other users of the programs.

hope it makes a bit of sense :stuck_out_tongue_winking_eye: 

We did have this in the mock-up, but as the **project name** now is given in spyder's `create new project` dialog we are in a 'post-factum' position, with no means to warn the user that the name is not permitted. (for our plugin) :frowning_face:

We discussed how we could handle such case, and the resolution was that the plugin (can) provide a regex (that outlines VALID names) and a 'fail message'. This way the project type plugins don't have to jump to awkward hoops (remember, the project directory and the .spyproject directory are already created whenever the plugin receives control) to get the message to the user.

Reply:
> hope it makes a bit of sense

Yes, it does now. Thanks for the explanation.